lib/attrsets.nix

```
@@ -1200,12 +1232,12 @@ rec {
     f:
     set:
     let
-      recurse = path:
-        mapAttrs
-          (name: value:
-            if isAttrs value && cond value
-            then recurse (path ++ [ name ]) value
-            else f (path ++ [ name ]) value);
+      recurse = path: value:
+        if isAttrs value && cond value
+        then mapAttrs (n: v: recurse (path ++ [n]) v) value
+        else if isList value && cond value
+        then imap0 (i: v: recurse (path ++ [(toString i)]) v) value
```

```suggestion
        then imap0 (i: v: recurse (path ++ [i]) v) value
```

Otherwise we can't distinguish between `{ "0" = 1; }` and `[ 1 ]` when recursing, because both would have `"0"` in the path. Passing through the integer without converting to a string makes it clear that integer path entries come from lists.